cryptography/src/lib.rs

```diff
@@ -417,27 +435,27 @@ mod tests {
         let mut hasher = H::new();
         hasher.update(b"hello world");
         let digest = hasher.finalize();
-        assert!(H::validate(&digest));
-        assert_eq!(digest.len(), H::len());
+        assert!(H::Digest::try_from(digest.as_ref()).is_ok());
```

Comment:
If we add `Deref<Target = [u8]>`, you can just do `&digest` here.

Reply:
Huum it is the only thing I did not change after implementing `Deref` as `&digest` fails. I get an error 
```
the trait bound `&<H as Hasher>::Digest: Into<<H as Hasher>::Digest>` is not satisfied
required for `&<H as Hasher>::Digest` to implement `Into<<H as Hasher>::Digest>`
```

But the deref works when trying to assign to a `&[u8]` ( ex: `let slice: &[u8] = &digest` )

Reply:
I took a look at this and didn't realize what was being tested. What you have currently is good (and we should still keep Deref) 👍